subprojects/base-services/src/main/java/org/gradle/internal/service/ServiceRegistry.java

```diff
@@ -36,7 +36,7 @@
      * @throws UnknownServiceException When there is no service of the given type available.
      * @throws ServiceLookupException On failure to lookup the specified service.
      */
-    @UsedByScanPlugin
+    @UsedByScanPlugin //and test-retry-plugin
```

Comment:
not sure how we should comment this to avoid any misunderstanding that actuall bot the scan-plugin and the test-retry-plugin are using this api.

Reply:
When I did this for the test-distribution plugin I left preexisting annotations unchanged.

Reply:
How about `@UsedByScanPlugin("scan, test-retry")`?